### ISSUE TITLE
Can't get code example on Fabric to work

### ISSUE THREAD

**Issue by alecb100** [TelepathicGrunt/StructureTutorialMod#26]
I am attempting to implement code checks into my custom structure to have more control over the spawning of my structure but no matter what I do I can't get it to run. I've followed your code example with the Sky Structures but when I get to editing my json file in the worldgen/structure folder and remove use_expansion_hack like it states in your json, the game crashes because it's not provided. When I return the key and open a new world, the checks that I did in the code seemingly are ignored.

I poured through vanilla structure code and tried searching up other solutions online but I can't seem to get my structure json to be untied to the key requirements that minecraft has set and instead use my own custom code with the codec. I'm not sure if 1.20.4 is just very different from 1.20.2 in this way or I am just doing something very wrong. Any help would be appreciated.

**Comment by TelepathicGrunt** [comment]
Without the codec, json, and log files, there isn’t anything I can say for advice. The json did not change except for addition of pool alias https://misode.github.io/worldgen/structure/

**Comment by alecb100** [comment]
The codec, and by extension the code, is copied straight from your SkyStructures.java class for now just to get things working, with the exception of making sure the height is below -30 in the extraSpawningChecks() just to verify the method is working. I can send my json files, logs don't have anything to say about my structure spawning, at least nothing that I could find, but I can still send those if you'd like.

```fates_mausoleum.json:
{
  "type": "minecraft:jigsaw",
  
  "start_pool": "mythos-amythologymod:fates_mausoleum/start_pool",
  
  "size": 1,
  
  "max_distance_from_center": 80,
  
  "biomes": "#mythos-amythologymod:has_structure/fates_mausoleum",
  
  "step": "surface_structures",
  
  "start_height": {
    "type": "minecraft:uniform",
    "min_inclusive": {
      "absolute": -25
    },
    "max_inclusive": {
      "absolute": 40
    }
  },
  
  "use_expansion_hack": false,
  "spawn_overrides": {},
  "values": [

  ]
}
```

**Comment by alecb100** [comment]
I couldn't tell if the code was working so I made the start_height what it is now and checking if it is below -30 and only letting it spawn if so just to verify the code is working. The structure still spawns above -30. I added this check because at first I was changing the startY integer in the getStructurePosition() method and noticing no changes either. I want to eventually get it so that most of the information is generated in code rather than json files, particularly the height, but I'm getting no where.
